cmd/assetsvc/postgresql_utils.go

```diff
@@ -235,34 +213,77 @@ func (m *postgresAssetManager) getChartFilesWithFallback(namespace, filesID stri
 	return chartFiles, nil
 }
 
-func containsVersionAndAppVersion(chartVersions []models.ChartVersion, version, appVersion string) (models.ChartVersion, bool) {
-	for _, ch := range chartVersions {
-		if ch.Version == version && ch.AppVersion == appVersion {
-			return ch, true
-		}
+func (m *postgresAssetManager) getPaginatedChartListWithFilters(cq chartQuery, pageNumber, pageSize int) ([]*models.Chart, int, error) {
+	whereQuery, whereQueryParams := m._generateWhereClause(cq)
```

Comment:
Now the logic for applying filters in the WHERE clause is shared by using `m._generateWhereClause`